### PR TITLE
Support multiple authentication methods for S3

### PR DESCRIPTION
**Pull request by chancez**
**What this PR does**:

Updates minio to v7 and configures a credentials chain that allows using
multiple authentication methods for accessing S3. In particular this
enables IRSA
https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html.

**Checklist**
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by annanay25 — APPROVED** [review]
Fantastic work, LGTM and thank you @chancez! I just have a few requests -

- Could you please add a documentation section highlighting the different authentication methods that can be used with the `s3` backend? Ideally under `docs/tempo/website` so its easily accessible?
- We'll need a signed CLA to merge this PR

**Comment by chancez** [comment]
Ok rebased, I signed the CLA, and added some documentation indicating supported authentication methods for S3. I wasn't sure where the best place to put the docs were, but I'm glad to make updates to them.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,3 +7,4 @@
 * [ENHANCEMENT] Add command line flags for s3 credentials. [#308](https://github.com/grafana/tempo/pull/308)
 * [BUGFIX] Increase Prometheus `notfound` metric on tempo-vulture. [#301](https://github.com/grafana/tempo/pull/301)
 * [BUGFIX] Return 404 if searching for a tenant id that does not exist in the backend. [#321](https://github.com/grafana/tempo/pull/321)
+* [ENHANCEMENT] Support multiple authentication methods for S3 (IRSA, IAM role, static). [#320](https://github.com/grafana/tempo/pull/320)
```
[the remark this answers is not in the document]
Comment (chancez) [reply CHANGELOG.md:10]: Ahhh I was wondering if there was any ordering here, git history wasnt so obvious about how y'all picked.

File: docs/tempo/website/configuration/_index.md [hunk 2]
```diff
@@ -56,6 +56,15 @@ compactor:
...
 The storage block is used to configure TempoDB.
 
+For the s3 backend, the following authentication methods are supported:
+
+- AWS env vars (static AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY)
+- Static access key and secret credentials specified in `storage.trace.s3.access_key` and `storage.trace.s3.secret_key`
+- Minio env vars
+- AWS shared credentials config file
+- Minio shared credentials config file
+- AWS IAM (IRSA via WebIdentity, EC2 instance role)
```
[the remark this answers is not in the document]
Comment (chancez) [reply docs/tempo/website/configuration/_index.md:66]: I'm only familiar with S3, not even minio, so I can link out to the minio config docs for the minio options, but I think I'll skip GCS, if you want to tackle that in a separate PR?

[the remark this answers is not in the document]
Comment (chancez) [reply docs/tempo/website/configuration/_index.md]: I added some links and elaborated on the env-vars. Let me know if this works for you.

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (chancez) [reply CHANGELOG.md]: This should be fixed.

[the remark this answers is not in the document]
Comment (annanay25) [reply CHANGELOG.md]: Haha, yeah you'll find these instructions in the PR template checklist :)

File: docs/tempo/website/configuration/_index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (annanay25) [reply docs/tempo/website/configuration/_index.md]: Thank you, this works :)

**Pull request merged**